crates/chain/src/chain_data.rs

```diff
@@ -140,6 +188,12 @@ impl Default for BlockId {
     }
 }
 
+impl BlockAnchor for BlockId {
+    fn anchor_block(&self) -> BlockId {
+        *self
+    }
+}
```

Comment:
I think I misunderstood your definition of "anchor". I thought a tx can only be anchored in a block it is included in... Now this changes things...

Reply:
That's correct. It does not have to be in that block. I don't think we need to know the `confirmation_height` for anything. We only need to know the `anchor_block`. This could be the block it's in or it could be a child of that block.

Reply:
> I don't think we need to know the confirmation_height for anything.

I don't think I agree with this. Knowing which height a transaction is confirmed is a basic wallet feature. The only place that makes sense to store this information is alongside an anchor.

Reply:
> Knowing which height a transaction is confirmed is a basic wallet feature. 

Correct. It is not a basic feature of an `Anchor` though which is why it should not exist on this trait. We only need the `BlockId` of the anchor. The thing that implements `Anchor` may very well have this information (it almost certainly will) but an API to get this doesn't need to be on this triat.

Reply:
That makes sense, thanks!